src/main/java/com/es/phoneshop/service/impl/DefaultDosService.java

```diff
@@ -0,0 +1,30 @@
+package com.es.phoneshop.service.impl;
+
+import com.es.phoneshop.service.DosService;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DefaultDosService implements DosService {
+
+    private static final int MAX_REQUEST_COUNT = 20;
+    private Map<String, Integer> requestCount = Collections.synchronizedMap(new HashMap<>());
```

Comment:
Method isAllowed is already synchronized. Usage of synchronizedMap is not required here